contracts/modules/cover/Cover.sol

```diff
@@ -781,20 +799,29 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     return uint(activeCover[assetId].totalActiveCoverInAsset);
   }
 
-  function isPoolAllowed(uint productId, uint poolId) external view returns (bool) {
-
-    uint poolCount = allowedPools[productId].length;
+  // Returns true if the product exists and the pool is authorized to have the product
+  function isPoolAllowed(uint productId, uint poolId) public view returns (bool) {
 
-    if (poolCount == 0) {
-      return true;
-    }
+    // If product exists and is not deprecated, check if the pool is allowed
+    if (productId < _products.length && !_products[productId].isDeprecated){
```

Comment:
as suggested by @shark0der 

this logic of product existence should be out of this function pulled in the outer calling function

```the product existence check should not happen inside isPoolAllowed but in the createStakingPool function (directly inside the for loop that calls isPoolAllowed seems to be a good place)```

or the function renamed